Update custom-proxy.md (#4243)
segmentio/segment-docs: src/connections/sources/catalog/libraries/website/javascript/custom-proxy.md

> info ""
> The **Host Address** field does not appear in source settings until it's enabled by Segment Customer Success.

**Option 2**

## Custom CDN / API Proxy

You can proxy settings and destination requests that typically go to `http://cdn.segment.com` through a custom proxy.

```ts

> const analytics = AnalyticsBrowser.load({

Reviewer: @silesky where can I get the instructions for snippet users? Can you please provide?

> const analytics = AnalyticsBrowser.load({

Reviewer 2: In the [snippet](https://segment.com/docs/getting-started/02-simple-install/) (typically in the `<head />` of your page, replace the following lines:
1.)
```js
t.src="https://cdn.segment.com/analytics.js/v1/" + key + "/analytics.min.js" // old
// with
t.src="https://MY-CUSTOM-CDN-PROXY.com/analytics.js/v1/" + key + "/analytics.min.js" // new
```
2.)
```js
analytics.load("<MY_WRITE_KEY>") // old
// with
analytics.load("<MY_WRITE_KEY>", { // new
    integrations: {
      "Segment.io": {
        apiHost:  "MY-CUSTOM-API-PROXY.com" // omit the https, e.g. "analytics.foo.com" rather than "https://analytics.foo.com/" 
      }
    }})
```

  writeKey,
  // GET http://cdn.segment.com/v1/projects/<writekey>/settings ->
  // https://MY-CUSTOM-CDN-PROXY.com/v1/project/<writekey>/settings
  // GET https://cdn.segment.com/next-integrations/actions/...js ->
  // https://MY-CUSTOM-CDN-PROXY.com/next-integrations/actions/...js
  cdnURL: 'https://MY-CUSTOM-CDN-PROXY.com' //
 })
```

> You can proxy event calls that typically go to `https://api.segment.io` by configuring `integrations['Segment.io'].apiHost`.

Reviewer 2: ```suggestion
You can proxy tracking calls that typically go to `api.segment.io/v1` by configuring `integrations['Segment.io'].apiHost`.
```

```ts
const analytics = AnalyticsBrowser.load(
    {
      writeKey,
      cdnURL: 'https://MY-CUSTOM-CDN-PROXY.com'
    },
    {
      integrations: {
        'Segment.io': {

>           // POST https://api.segment.io/v1/t ->

Reviewer 2: ```suggest
// POST https://api.segment.io/v1/t -> https://MY-CUSTOM-API-PROXY.com/t

```

>           // POST https://api.segment.io/v1/t ->
>           //  https://MY-CUSTOM-API-PROXY.com/v1/t

Reviewer 2: ```suggestion
          // POST https://api.segment.io/v1/t -> https://MY-CUSTOM-API-PROXY.com/t
```

>           apiHost: 'MY-CUSTOM-API-PROXY.com' //

Reviewer 2: There's a stray '//' at the end of the line... we might want to add a comment like:
```
// omit the https, e.g. "analytics.foo.com" rather than "https://analytics.foo.com/" 
```

>           apiHost: 'MY-CUSTOM-API-PROXY.com' //

Reviewer 2: ```suggestion
          apiHost: 'MY-CUSTOM-API-PROXY.com' // omit the protocol (http/https) e.g. "api.segment.io/v1" rather than "https://api.segment.io/v1" 
```

> info ""
> The **Host Address** field does not appear in source settings until it's enabled by Segment Customer Success.

## Custom CDN / API Proxy

> 

Reviewer 2: ```suggestion
### [`npm`](https://www.npmjs.com/package/@segment/analytics-next) library-users instructions
```

  cdnURL: 'https://MY-CUSTOM-CDN-PROXY.com' 
 })
```

Proxy tracking calls that typically go to `api.segment.io/v1` by configuring `integrations['Segment.io'].apiHost`.
```ts
const analytics = AnalyticsBrowser.load(
    {
      writeKey,
      cdnURL: 'https://MY-CUSTOM-CDN-PROXY.com'
    },
    {
      integrations: {
        'Segment.io': {
          // POST https://api.segment.io/v1/t -> https://MY-CUSTOM-API-PROXY.com/t
          apiHost: 'MY-CUSTOM-API-PROXY.com' // omit the protocol (http/https) e.g. "api.segment.io/v1" rather than "https://api.segment.io/v1" 
        }
      }
    }
  )
```

### Snippet instructions
If you're a snippet user, you need to modify the [analytics snippet](/docs/getting-started/02-simple-install/#step-1-copy-the-snippet) that's inside your `<head>`. 

To proxy settings and destination requests that typically go to `http://cdn.segment.com`, replace: 
```diff
- t.src="https://cdn.segment.com/analytics.js/v1/" + key + "/analytics.min.js"
+ t.src="https://MY-CUSTOM-CDN-PROXY.com" + key + "/analytics.min.js"
```

To proxy tracking calls that typically go to `api.segment.io/v1`, replace:
```diff
- analytics.load("<MY_WRITE_KEY>")

> + analytics.load("<MY_WRITE_KEY>", { integrations: { "Segment.io": { apiHost:  "MY-CUSTOM-API-PROXY.com" }}})

Reviewer 2: cc @pooyaj, @chrisradek --  we have a place in the UI where we also allow apiHost to be overridden for custom domain customers -- if so, the "apiHost" configuration might  not be necessary here?

Reviewer 2: @prigiattiperrut -- it looks like we want to push customers towards using the UI. can we remove the feature flag here: https://github.com/segmentio/app/blob/77ff57aac27a98f207ea6bc828a118e0d5bb3340/client/components/Sources/Settings/AnalyticsJs.tsx#L485-L523?

Author: Hmm! Sorry, I will have to ask a few questions in return, just to clarify:

1.  How often do we need to figure out which sources are proxying Segment?
2. Do we have an ETA for when the apiHost UI will be available to all customers by default? 

I think if the answer to #1 is not very often and #2 we don't have an ETA, it will take some time, then I believe that we could still make this information available to customers as they will benefit from knowing it.

Reviewer 2: > How often do we need to figure out which sources are proxying Segment?
This is a nice-to-have -- 
> Do we have an ETA for when the apiHost UI will be available to all customers by default?
I have no idea -- that's not my team =) I was hoping you knew who was in charge (or if it's up to us to make the PR)

Reviewer 3: I agree we should push the update rather than wait for the UI.
